src/Cms/PageCopy.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Uuid\Uuid;
+use Kirby\Uuid\Uuids;
+
+/**
+ * Normalizes a newly generated copy of a page,
+ * adapting page slugs, UUIDs etc.
+ * (for single as well as multilang setups)
+ *
+ * @package   Kirby Cms
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.0.0
+ * @internal
+ */
+class PageCopy
+{
+	public function __construct(
+		public Page $copy,
+		public Page|null $original = null,
+		public bool $files = false,
+		public bool $children = false,
+		public array $uuids = []
+	) {
+	}
+
+	/**
+	 * Returns the pages collection for children to adapt
+	 */
+	public function children(): Pages
+	{
+		return match ($this->children) {
+			true  => $this->copy->index(drafts: true),
+			false => new Pages()
+		};
```

Comment:
```suggestion
		if ($this->children === true) {
			return $this->copy->index(drafts: true);
		}

		return new Pages();
```

I can't help but feel uneasy with using match everywhere to be honest. It's one of those decisions that we would need to move along a lot more in so many other parts to turn it into a consistent coding style and I'm really not sure if it is helping. I don't want to block PRs because of this. I really just think that we should decide at which point it makes sense to use match and when there are just two cases like here, I would vote against it.